aiida/orm/querybuilder.py

```diff
@@ -2095,16 +2127,27 @@ def distinct(self) -> 'QueryBuilder':
         self._query = self.get_query().distinct()
         return self
 
-    def first(self) -> Optional[List[RowType]]:
+    def analyze_query(self, execute: bool = True, verbose: bool = False) -> str:
+        """Return the query plan, i.e. a list of SQL statements that will be executed.
+
+        See: https://www.postgresql.org/docs/11/sql-explain.html
+
+        :params execute: Carry out the command and show actual run times and other statistics.
+        :params verbose: Display additional information regarding the plan.
         """
-        Executes query asking for one instance.
-        Use as follows::
+        query = self.get_query()
+        if query.session.bind.dialect.name != 'postgresql':
+            raise NotImplementedError('Only PostgreSQL is supported for this method')
+        compiled = self._compile_query(query, literal_binds=True)
+        options = ', '.join((['ANALYZE'] if execute else []) + (['VERBOSE'] if verbose else []))
+        options = f' ({options})' if options else ''
+        rows = self._impl.get_session().execute(f'EXPLAIN{options} {compiled.string}').fetchall()
+        return '\n'.join(row.values()[0] for row in rows)
 
-            qb = QueryBuilder(**queryhelp)
-            qb.first()
+    def first(self) -> Optional[List[RowType]]:
+        """Executes the query, asking for the first row of results.
```

Comment:
```suggestion
        """Executes the query, asking for the first row of results.

        Note, this may change if several rows are valid for the query,
        as persistent ordering is not guaranteed unless explicitly specified.
```